packages/eslint-plugin/rules/i18n-no-collapsible-whitespace.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTextContentFromNode,
+	getTranslateFunctionName,
+} = require( '../utils' );
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		schema: [],
+		messages: {
+			noCollapsibleWhitespace:
+				'Translations should not contain collapsible whitespace{{problem}}',
+		},
+	},
+	create( context ) {
+		return {
+			CallExpression( node ) {
+				const { callee, arguments: args } = node;
+
+				const functionName = getTranslateFunctionName( callee );
+
+				if ( ! TRANSLATION_FUNCTIONS.includes( functionName ) ) {
```

Comment:
Minor-ish (it's a relatively hot code path): `Set#has` is arguably more semantically correct in our usage, and marginally more performant.

https://jsperf.com/set-has-vs-array-includes